precice.f03

```diff
@@ -38,458 +31,284 @@ subroutine precicef_advance(timestepLengthLimit) &
-      integer(kind=c_int) :: hasToEvaluate
-    end subroutine precicef_has_to_evaluate_fine_model        
-
-    subroutine precicef_is_action_required(action, isRequired, lengthAction) &
-      &  bind(c, name='precicef_is_action_required_')
-
-      use, intrinsic :: iso_c_binding
-      character(kind=c_char), dimension(*) :: action
-      integer(kind=c_int)                  :: isRequired
-      integer(kind=c_int), value           :: lengthAction
-    end subroutine precicef_is_action_required
-
-    ! Deprecated - Forwards to precicef_is_action_required_
-    subroutine precicef_action_required(action, isRequired, lengthAction) &
-      &  bind(c, name='precicef_is_action_required_')
-
-      use, intrinsic :: iso_c_binding
-      character(kind=c_char), dimension(*) :: action
-      integer(kind=c_int)                  :: isRequired
-      integer(kind=c_int), value           :: lengthAction
-    end subroutine precicef_action_required
-
-    subroutine precicef_mark_action_fulfilled(action, lengthAction) &
-      &  bind(c, name='precicef_mark_action_fulfilled_')
-
-      use, intrinsic :: iso_c_binding
-      character(kind=c_char), dimension(*) :: action
-      integer(kind=c_int), value           :: lengthAction
-    end subroutine precicef_mark_action_fulfilled
-
-    subroutine precicef_has_mesh(meshName, hasMesh, lengthMeshName) &
-      &  bind(c, name='precicef_has_mesh_')
+    subroutine precicef_get_max_time_step_size(maxTimeStepSize) &
+      & bind(c, name='precicef_get_max_time_step_size_')
 
       use, intrinsic :: iso_c_binding
-      character(kind=c_char), dimension(*) :: meshName
-      integer(kind=c_int)                  :: hasMesh
-      integer(kind=c_int), value           :: lengthMeshName
-    end subroutine precicef_has_mesh  
+      real(kind=c_double) :: maxTimeStepSize
+    end subroutine precicef_get_max_time_step_size
 
-    subroutine precicef_get_mesh_id(meshName, meshID, lengthMeshName) &
-      &  bind(c, name='precicef_get_mesh_id_')
+    subroutine precicef_requires_mesh_connectivity_for(meshName, required, meshNameLength) &
+      & bind(c, name='precicef_requires_mesh_connectivity_for_')
 
       use, intrinsic :: iso_c_binding
       character(kind=c_char), dimension(*) :: meshName
-      integer(kind=c_int)                  :: meshID
-      integer(kind=c_int), value           :: lengthMeshName
-    end subroutine precicef_get_mesh_id
+      integer(kind=c_int) :: required
+      integer(kind=c_int), value :: meshNameLength
+    end subroutine precicef_requires_mesh_connectivity_for
 
-    subroutine precicef_set_vertex(meshID, position, vertexID) &
+    subroutine precicef_set_vertex(meshName, position, vertexID) &
       &  bind(c, name='precicef_set_vertex_')
 
       use, intrinsic :: iso_c_binding
-      integer(kind=c_int) :: meshID
-      real(kind=c_double) :: position(3)
-      integer(kind=c_int) :: vertexID
+      character(kind=c_char), dimension(*) :: meshName
```

Comment:
Length is missing

Reply:
Good catch, fixed, thanks!

I guess we need kind of an "extended solver dummy" that only calls all functions to make sure that they are all correct and link to preCICE.

Or somehow check the linking during compilation of the module.

Reply:
The mocked interface (workpackage of preECO) should ultimately be a solution for this problem. Then, such a extended solverdummy can run alone and also test the API, i.e. it becomes a normal test "main".

Reply:
I had a simpler usecase in mind (just that the interface is correct), which would probably be solvable in a simpler way, but having a mocked interface would also of course help.

Note that the problem here is not that we cannot call preCICE. It is that the building procedure of the interface does not try to link to preCICE.

Reply:
Yes, I get that. With the mocked interface, we would then test the linking when running the tests. I am not sure if there is a simpler solution?

Reply:
Problem with a solverdummy is that you need to use all API functions in a meaningful "integration-test-like" setting. This gets complicated ...
With the mocked interface, we could test each API function in an isolated setup.

Reply:
But not if we don't ever run the built executable, right?

Reply:
If the linking fails when the mod is incorrect, then a mocked interface makes sense. Is that the case though?

Reply:
Let's continue the discussion in this issue: https://github.com/precice/fortran-module/issues/15